axelor-bank-payment/src/main/java/com/axelor/apps/bankpayment/service/batch/BatchCreditTransferSupplierPaymentBankPayment.java

```diff
@@ -51,16 +60,48 @@ public BatchCreditTransferSupplierPaymentBankPayment(
 
   @Override
   protected void process() {
-    List<InvoicePayment> doneList =
-        processInvoices(InvoiceRepository.OPERATION_TYPE_SUPPLIER_PURCHASE);
+    if (!stopping) {
+      List<InvoicePayment> doneList =
+          processInvoices(InvoiceRepository.OPERATION_TYPE_SUPPLIER_PURCHASE);
 
-    if (!doneList.isEmpty()) {
-      try {
-        bankOrderMergeService.mergeFromInvoicePayments(doneList);
-      } catch (Exception e) {
-        TraceBackService.trace(e, ExceptionOriginRepository.INVOICE_ORIGIN, batch.getId());
-        LOG.error(e.getMessage());
+      if (!doneList.isEmpty()) {
+        try {
+          bankOrderMergeService.mergeFromInvoicePayments(doneList);
+        } catch (Exception e) {
+          TraceBackService.trace(e, ExceptionOriginRepository.INVOICE_ORIGIN, batch.getId());
+          LOG.error(e.getMessage());
+        }
       }
     }
   }
+
+  @Override
+  protected void start() throws IllegalAccessException {
+    super.start();
+
+    try {
+      this.testCreditTransferBatchField();
+    } catch (AxelorException e) {
+      TraceBackService.trace(
+          new AxelorException(e, e.getCategory(), ""),
+          ExceptionOriginRepository.CREDIT_TRANSFER,
+          batch.getId());
+      incrementAnomaly();
+      stopping = true;
+    }
+    checkPoint();
+  }
+
+  public void testCreditTransferBatchField() throws AxelorException {
+    AccountingBatch accountingBatch = batch.getAccountingBatch();
+    if (ObjectUtils.isEmpty(accountingBatch.getBankDetails())
+        && accountingBatch.getPaymentMode().getTypeSelect() == PaymentModeRepository.TYPE_TRANSFER
+        && accountingBatch.getPaymentMode().getInOutSelect() == PaymentModeRepository.OUT) {
```

Comment:
can this cause a NPE if payment mode is missing ?

Reply:
Resolved